tests/plenary/path_spec.lua

```diff
@@ -205,6 +205,89 @@ describe('Path', function()
     end)
   end)
 
+  describe('rename', function()
+    it('can rename a file', function()
+      local p = Path:new("a_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(pcall(p.rename, p, { new_name = "not_a_random_filename.lua" }))
+      assert.are.same("not_a_random_filename.lua", p.filename)
+
+      p:rm()
+    end)
+
+    it('can handle an invalid filename', function()
+      local p = Path:new("some_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(not pcall(p.rename, p, { new_name = "" }))
+      assert(not pcall(p.rename, p))
+      assert.are.same("some_random_filename.lua", p.filename)
+
+      p:rm()
+    end)
+
+    it('can move to parent dir', function()
+      local p = Path:new("some_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(pcall(p.rename, p, { new_name = "../some_random_filename.lua" }))
+      assert.are.same("some_random_filename.lua", p.filename)
+
+      p:rm()
+    end)
+
+    it('cannot rename to an existing filename', function()
+      local p1 = Path:new("a_random_filename.lua")
+      local p2 = Path:new("not_a_random_filename.lua")
+      assert(pcall(p1.touch, p1))
+      assert(pcall(p2.touch, p2))
+      assert(p1:exists())
+      assert(p2:exists())
+
+      assert(not pcall(p1.rename, p1, { new_name = "not_a_random_filename.lua" }))
+      assert.are.same(p1.filename, "a_random_filename.lua")
+
+      p1:rm()
+      p2:rm()
+    end)
+  end)
+
+  describe('copy', function()
```

Comment:
You can add another test for copy to different directory, just to have it complete